.github/workflows/workflow.yaml

```diff
@@ -139,6 +139,8 @@ jobs:
               ./tests/run-start-script.sh --use-nix --with-bzlmod=${{ matrix.bzlmod }}
             fi
             bazel build //tests:run-tests
+            # Shutdown Bazel to free up memory
+            bazel shutdown
```

Comment:
That might already help, but `RunTest.hs` is invoking bazel again and starts a few server instances. Maybe we should also add shutdown calls there...

Reply:
@avdv Do you mean after `run-tests` has completed but before it returns to this script?

Reply:
I mean here:
https://github.com/tweag/rules_haskell/blob/c2c6d3f7a5f51b49b65d98ede37f20914e591475/rules_haskell_tests/tests/RunTests.hs#L153-L156
and here:
https://github.com/tweag/rules_haskell/blob/c2c6d3f7a5f51b49b65d98ede37f20914e591475/rules_haskell_tests/tests/RunTests.hs#L149-L151

The spec items are run sequentially and all the items before these lines are run inside the current workspace, but these items run in a different workspace.

Reply:
I added a task to #2089.